mbed/mbed.py

```diff
@@ -1636,7 +1641,7 @@ def check_requirements(self, show_warning=False):
                     if not pkg in installed_packages:
                         missing.append(pkg)
 
-                if missing and install_requirements:
+                if missing and install_requirements and not show_warning_only:
```

Comment:
Oh hang on, so now the `show_warning_only` is being used to select if things should be auto installed? That doesn't sound right.

Reply:
show_warning_only = do not auto-install

Reply:
@screamerbg Perhaps it could use a different name then?

Reply:
@theotherjimmy Proposal?

Reply:
`require_install`?and invert the logic so that True raises an error and False warns.